pkg/sources/github/graphql.go

```diff
@@ -0,0 +1,491 @@
+package github
+
+import (
+	"cmp"
+	"fmt"
+	"math/rand/v2"
+	"strings"
+	"time"
+
+	"github.com/shurcooL/githubv4"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/source_metadatapb"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/sanitizer"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/sources"
+)
+
+// processIssuesWithComments process github repo issues with comments using graphql API
+func (s *Source) processIssuesWithComments(
+	ctx context.Context, repoInfo repoInfo,
+	reporter sources.ChunkReporter, cutoffTime *time.Time,
+) error {
+	vars := map[string]any{
+		owner:              githubv4.String(repoInfo.owner),
+		repository:         githubv4.String(repoInfo.name),
+		issuesPerPage:      githubv4.Int(defaultPagination),
+		issuesPagination:   (*githubv4.String)(nil),
+		commentsPerPage:    githubv4.Int(defaultPagination),
+		commentsPagination: (*githubv4.String)(nil),
+	}
+
+	var totalIssues int
+
+	// loop will continue as long as there are issues in the repository
+	for {
+		var query issuesWithComments
+		err := s.connector.GraphQLClient().Query(ctx, &query, vars)
+		if s.handleGraphqlRateLimitWithChunkReporter(ctx, reporter, &query.RateLimit, err) {
+			continue
+		}
+
+		if err != nil {
+			return fmt.Errorf("error fetching issues: %w", err)
+		}
+
+		totalIssues += len(query.GetIssues())
+
+		ctx.Logger().V(5).Info("Scanning Issues",
+			"total_issues", len(query.GetIssues()))
+
+		if err := s.chunkGraphqlIssues(ctx, repoInfo, query.Repository.Issues.Nodes, reporter); err != nil {
+			return err
+		}
+
+		// process each issue comments
+		for _, issue := range query.GetIssues() {
+			ctx.Logger().V(5).Info("Scanning Issue Comments",
+				"issue_id", issue.Number,
+				"total_comments", len(issue.GetIssueComments()),
+			)
+
+			if err := s.chunkComments(ctx, repoInfo, issue.GetIssueComments(), reporter, cutoffTime); err != nil {
+				return err
+			}
+
+			// if issue has more than 100 comments, we need to send another request for that specific issue to pull more comments
+			for issue.Comments.PageInfo.HasNextPage {
+				commentVars := map[string]any{
+					owner:              githubv4.String(repoInfo.owner),
+					repository:         githubv4.String(repoInfo.name),
+					issueNumber:        githubv4.Int(issue.Number),
+					commentsPerPage:    githubv4.Int(defaultPagination),
+					commentsPagination: issue.Comments.PageInfo.EndCursor,
+				}
+
+				// request this issue more comments
+				var commentsQuery singleIssueComments
+				err := s.connector.GraphQLClient().Query(ctx, &commentsQuery, commentVars)
+				if s.handleGraphqlRateLimitWithChunkReporter(ctx, reporter, &query.RateLimit, err) {
```

Comment:
```suggestion
				if s.handleGraphqlRateLimitWithChunkReporter(ctx, reporter, &commentsQuery.RateLimit, err) {
```

Not sure if this is what you want or not.

Reply:
Good catch 👀